darwin/dataset/download_manager.py

```diff
@@ -95,11 +95,13 @@ def download_all_images_from_annotations(
 
         if not force_replace:
             # Check the planned path for the image against the existing images
-            planned_image_path = (
-                images_path
-                / Path(annotation.remote_path.lstrip("/\\")).resolve().absolute()
-                / Path(annotation.filename)
-            )
+            filename = Path(annotation.slots[0].source_files[0]["file_name"])
+            if use_folders and annotation.remote_path != "/":
+                planned_image_path = (
+                    images_path / Path(annotation.remote_path.lstrip("/\\")) / filename
+                )
+            else:
+                planned_image_path = images_path / filename
```

Comment:
When downloading a release, darwin-py skips files that have already been downloaded in previous `pull()` operations by checking `planned_image_path` against the local filesystem. These changes fix a bug whereby `planned_image_path` only worked for files that were downloaded with folders